app/src/main/kotlin/com/wire/android/services/ServicesManager.kt

```diff
@@ -25,26 +25,109 @@ import android.content.Context
 import android.content.Intent
 import android.os.Build
 import com.wire.android.appLogger
-import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.android.di.KaliumCoreLogic
+import com.wire.android.util.dispatchers.DispatcherProvider
+import com.wire.kalium.logic.CoreLogic
+import com.wire.kalium.logic.data.team.Team
+import com.wire.kalium.logic.data.user.SelfUser
 import com.wire.kalium.logic.data.user.UserId
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.debounce
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.distinctUntilChangedBy
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import org.jetbrains.annotations.VisibleForTesting
 import javax.inject.Inject
+import javax.inject.Singleton
 import kotlin.reflect.KClass
 
 /**
  * This is helper class that should be used for starting/stopping any services.
  * The idea is that we don't want to inject, or provide any context into ViewModel,
  * but to have an ability start Service from it.
  */
-class ServicesManager @Inject constructor(private val context: Context) {
+@Singleton
+class ServicesManager @Inject constructor(
+    private val context: Context,
+    dispatcherProvider: DispatcherProvider,
+    @KaliumCoreLogic private val coreLogic: CoreLogic,
+) {
+    private val scope = CoroutineScope(SupervisorJob() + dispatcherProvider.default())
+    private val ongoingCallServiceForUsers = MutableStateFlow<List<OngoingCallData>>(emptyList())
+
+    suspend fun currentlyOngoingCall(): Flow<OngoingCallData?> = ongoingCallServiceForUsers
+        .combine(
+            coreLogic.getGlobalScope().observeValidAccounts()
+        ) { ongoingCalls: List<OngoingCallData>, validAccounts: List<Pair<SelfUser, Team?>> ->
+            ongoingCalls.lastOrNull { ongoingCallData ->
+                // filter out calls that are for accounts that are not valid anymore
+                validAccounts.any { (selfUser, _) -> selfUser.id == ongoingCallData.userId }
+            }
+        }
```

Comment:
Yep, but `ObserveOngoingCallsUseCase` returns calls only for the specific user but we need to get info for all users, that's why there's this buffer, I renamed it to `ongoingCallsData`

Reply:
Yep, but ObserveOngoingCallsUseCase returns calls only for the specific user but we need to get info for all users, that's why there's this buffer, I renamed it to ongoingCallsData

Reply:
Yes `ObserveOngoingCallsUseCase` will observe ongoing calls for the current selected user and that's what we need I would say, we can't have ongoing calls for other users

Reply:
Ok, but then, why did the app already check the ongoing calls for all valid users? I just wanted to keep the same logic, so maybe we already had it wrong. Are you sure that the app always switches the user when the call becomes ongoing so that we can rely on only checking it for the currently selected user? 🤔

Reply:
I didn't work on that part, maybe we check valid users to end the call on logout.
And yes we do switch the the user when accepting a call on non current user

Reply:
Ok, it took me some time as I had to move some tests around and update them, but I changed and simplified it so that now it only observes ongoing calls for current user

Reply:
Thank you for applying the changes